client/sites/tools/logs/index.tsx

```diff
@@ -1,253 +1,332 @@
+						severity_deprecated: severity.includes( 'Deprecated' ),
+						severity_fatal: severity.includes( 'Fatal' ),
+					} )
+				);
+			}
 
-	const handleRequestTypeChange = ( requestType: string ) => {
-		setRequestType( requestType );
-		setAutoRefresh( false );
-		updateFilterQueryParam( 'request_type', requestType );
-	};
+			// Disable auto-refresh if the user navigates to a different page.
+			if ( autoRefresh === true && newView.page !== view.page ) {
+				setAutoRefresh( false );
+			}
 
-	const handleRequestStatusChange = ( requestStatus: string ) => {
-		setRequestStatus( requestStatus );
-		setAutoRefresh( false );
-		updateFilterQueryParam( 'request_status', requestStatus );
-	};
+			setView( newView );
+		},
+		[ autoRefresh, setAutoRefresh, oldSeverity, view.page, setView, dispatch ]
+	);
 
-	const headerTitles =
-		logType === 'php'
-			? [ 'severity', 'timestamp', 'message' ]
-			: [ 'request_type', 'date', 'status', 'request_url' ];
+	const fields = useFields( { logType } );
+	const { data, paginationInfo, isLoading } = useData( {
+		view,
+		logType,
+		startTime,
+		endTime,
+	} );
+	const actions: Action< PHPLog | ServerLog >[] = useActions( { logType, isLoading } );
+
+	const { downloadLogs, state } = useSiteLogsDownloader( { roundDateRangeToWholeDays: false } );
+	const isDownloading = state.status === 'downloading';
+	const onDownloadLogs = useCallback( () => {
+		downloadLogs( {
+			logType,
+			startDateTime: startTime,
+			endDateTime: endTime,
+			filter: toFilterParams( { view, logType } ),
+		} );
+	}, [ downloadLogs, logType, startTime, endTime, view ] );
+
+	const [ itemDetailsModal, setItemDetailsModal ] = useState< PHPLog | ServerLog | null >( null );
+	const onOpenDetailsModal = useCallback( ( item: PHPLog | ServerLog ) => {
+		setItemDetailsModal( item );
+	}, [] );
+	const onCloseDetailsModal = useCallback( () => {
+		setItemDetailsModal( null );
+	}, [] );
 
 	return (
-		<div>
-			<SiteLogsHeader
-				endDateTime={ dateRange.endTime }
-				logType={ logType }
-				requestStatus={ requestStatus }
-				requestType={ requestType }
-				severity={ severity }
-				startDateTime={ dateRange.startTime }
-			/>
-
-			<div className="site-logs-container">
-				{ siteId && <QuerySiteSettings siteId={ siteId } /> }
-				<SiteLogsToolbar
-					onDateTimeChange={ handleDateTimeChange }
-					onSeverityChange={ handleSeverityChange }
-					onRequestTypeChange={ handleRequestTypeChange }
-					onRequestStatusChange={ handleRequestStatusChange }
-					onAutoRefreshChange={ handleAutoRefreshClick }
-					logType={ logType }
-					startDateTime={ dateRange.startTime }
-					endDateTime={ dateRange.endTime }
-					autoRefresh={ autoRefresh }
-					severity={ severity }
-					requestType={ requestType }
-					requestStatus={ requestStatus }
+		<>
+			{ siteId && <QuerySiteSettings siteId={ siteId } /> }
+			{ !! itemDetailsModal && (
+				<DetailsModal
+					item={ itemDetailsModal }
+					actions={ actions }
```

Comment:
I don't really get why we need to pass the `actions` here. Shouldn't the reusable modal just render things based on the `item` and be used (imported) here and at the action that uses it too?

Reply:
Applied the suggestion at https://github.com/Automattic/wp-calypso/pull/99056/commits/be8946dcb6412dc48eb9efef7809a6c78aefcfb9